pricegraph/src/orderbook.rs

```diff
@@ -140,6 +141,69 @@ impl Orderbook {
         debug_assert!(!self.is_overlapping());
     }
 
+    /// Reduces all overlapping orders for a given market and returns the
+    /// computed overlapping transitive orderbook for that market.
+    pub fn reduce_overlapping_transitive_orderbook(
+        &mut self,
+        base: TokenId,
+        quote: TokenId,
+    ) -> TransitiveOrderbook {
+        let (base, quote) = (node_index(base), node_index(quote));
+
+        let mut overlap = TransitiveOrderbook::default();
+        while let Err(NegativeCycle(predecessors, node)) =
+            bellman_ford::search(&self.projection, quote)
+        {
+            let path = {
+                let mut cycle = path::find_cycle(&predecessors, node, Some(quote))
+                    .expect("negative cycle not found after being detected");
+                cycle.push(cycle[0]);
+                cycle
+            };
+
+            if path[0] == quote {
+                if let Some(base_index) = path.iter().position(|node| *node == base) {
+                    let (ask, bid) = (&path[0..base_index + 1], &path[base_index..]);
+
+                    overlap.asks.push(
+                        self.find_transitive_order(ask)
+                            .expect("ask transitive path not found after being detected"),
+                    );
+                    overlap.bids.push(
+                        self.find_transitive_order(bid)
+                            .expect("bid transitive path not found after being detected"),
+                    );
+                }
+            }
+
+            self.fill_path(&path).unwrap_or_else(|| {
+                panic!(
+                    "failed to fill path along detected negative cycle {}",
+                    format_path(&path),
+                )
+            });
+        }
+
+        overlap
+    }
+
+    /// Computes the equivalent transient order along a path.
```

Comment:
```suggestion
    /// Computes the equivalent transitive order along a path.
```
I have infected you with this word :) .

Reply:
:rofl: